src/sickle.ts

```diff
@@ -399,6 +406,75 @@ class Annotator {
     }
   }
 
+  private visitExterns(node: ts.Node, namespace: string[] = []) {
+    this.currentOutput = this.externsOutput;
+    switch (node.kind) {
+      case ts.SyntaxKind.ModuleDeclaration:
+        let decl = <ts.ModuleDeclaration>node;
+        namespace = namespace.concat([decl.name.text]);
+        this.visitExterns(decl.body, namespace);
+        break;
+      case ts.SyntaxKind.ModuleBlock:
+        let block = <ts.ModuleBlock>node;
+        for (let stmt of block.statements) {
+          this.visitExterns(stmt, namespace);
+        }
+        break;
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this.writeExternsInterface(<ts.InterfaceDeclaration>node, namespace);
+        break;
+      case ts.SyntaxKind.VariableStatement:
+        for (let decl of(<ts.VariableStatement>node).declarationList.declarations) {
+          this.writeExternsVariable(decl, namespace);
+        }
+        break;
+      default:
+        this.errorUnimplementedKind(node, 'externs generation');
+        break;
+    }
+    this.currentOutput = this.tsOutput;
+  }
+
+  private writeExternsInterface(decl: ts.InterfaceDeclaration, namespace: string[]) {
+    this.emit('/** @record @struct */\n');
+    let className = namespace.concat([decl.name.getText()]).join('.');
+    if (namespace.length > 0) {
+      this.emit(`${className} = function() {};\n`);
+    } else {
+      this.emit(`function ${className}() {}\n`);
+    }
+    for (let member of decl.members) {
+      switch (member.kind) {
+        case ts.SyntaxKind.PropertySignature:
+          let prop = <ts.PropertySignature>member;
+          this.maybeEmitJSDocType(prop.type, '@type');
+          this.emit(`\n${className}.prototype.${prop.name.getText()};\n`);
+          break;
+        default:
+          // Members can include things like index signatures, for e.g.
+          //   interface Foo { [key: string]: number; }
```

Comment:
not related to the code review, but guess is that index signatures can be expressed in closure by implementing the IObject<string, number> interface.